### PR TITLE
Http client followup

### PR DESCRIPTION
**Pull request by alxhub**
## PR Checklist
Does please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: packages/common/http/src/request.ts [hunk 1]
```diff
@@ -116,37 +113,55 @@ export class HttpRequest<T> {
...
+   * Outgoing URL parameters.
+   */
+  readonly params: HttpUrlEncodedParams;
+
+  /**
+   * The URL before parameters.
+   */
+  private _url: string;
+
+  readonly url: string;
```
Comment (IgorMinar) [inline packages/common/http/src/request.ts]: add a doc plz

File: packages/common/http/src/request.ts [hunk 2]
```diff
@@ -244,9 +287,8 @@ export class HttpRequest<T> {
     if (typeof this.body === 'string') {
       return 'text/plain';
     }
-    // `HttpUrlEncodedBody` is detected specially so as to allow it to be
-    // tree-shaken.
-    if (isUrlEncodedBody(this.body)) {
+    // `HttpUrlEncodedParams` has its own content-type.
+    if (HttpUrlEncodedParams instanceof HttpUrlEncodedParams) {
```
Comment (IgorMinar) [inline packages/common/http/src/request.ts]: ?

**Review by IgorMinar — APPROVED** [review]
- [ ] Please rename HttpUrlEncodedParams to just HttpParams. 
- [ ] fix the instanceof typo and add a test for that code path
- [ ] add missing docs

otherwise lgtm

File: packages/common/http/src/request.ts (hunk 2 above)
Comment (alxhub) [inline packages/common/http/src/request.ts]: Fixed.

File: packages/common/http/src/request.ts (hunk 1 above)
Comment (alxhub) [inline packages/common/http/src/request.ts]: I returned `url` to the way it was, and added `urlWithParams` with documentation.

**Pull request merged**